drivers/spi/spi_nrfx_spim.c

```diff
@@ -218,15 +298,26 @@ static void transfer_next_chunk(const struct device *dev)
 		xfer.p_rx_buffer = ctx->rx_buf;
 		xfer.rx_length   = spi_context_rx_buf_on(ctx) ? chunk_len : 0;
 
-		/* This SPIM driver is only used by the NRF52832 if
-		   SOC_NRF52832_ALLOW_SPIM_DESPITE_PAN_58 is enabled */
-		if (IS_ENABLED(CONFIG_SOC_NRF52832) &&
-		   (xfer.rx_length == 1 && xfer.tx_length <= 1)) {
-			LOG_WRN("Transaction aborted since it would trigger nRF52832 PAN 58");
-			error = -EIO;
-		}
-
-		if (!error) {
+#ifdef CONFIG_SOC_NRF52832_ALLOW_SPIM_DESPITE_PAN_58
+		if ((xfer.rx_length == 1 && xfer.tx_length <= 1)) {
+			if (dev_config->pan_58_work_around) {
+				setup_workaround_for_ftpan_58(dev_config->spim.p_reg,
+							      dev_data->ppi_ch,
+							      dev_data->gpiote_ch);
+				result = nrfx_spim_xfer(&dev_config->spim, &xfer, 0);
+				clear_workaround_for_ftpan_58(dev_data->ppi_ch,
+							      dev_data->gpiote_ch);
```

Comment:
Ok I see what you are saying here, I'll fix.

Reply:
In commit https://github.com/zephyrproject-rtos/zephyr/commit/57064fd93d1b370a2d567b899eb03333702b1d46 I moved `clear_workaround_for_ftpan_58()` to the end of `transfer_next_chunk()` as it will be called when there is no more data left to send or if `nrfx_spim_xfer()` returns with a failure for some reason.  Please let me know if you can see a problem with this.

Reply:
I think it won't work for example for a transfer like:
```
	const struct spi_buf rx_bufs[] = {
		{
			.buf = buf1,
			.len = 1,
		},
		{
			.buf = buf2,
			.len = 2,
		},
	};
	const struct spi_buf_set rx = {
		.buffers = rx_bufs,
		.count = 2,
	};
```
IMO `clear_workaround_for_ftpan_58()` should be called after each chunk transfer is started with `nrfx_spim_xfer()`. That's why I suggested to move it to `event_handler()`. You could skip disabling and re-enabling the workaround if you know that the next chunk is also 1 byte long with RX involved, but I'm not sure if it's worth complicating the logic.

Also I think it would be better to call `nrfx_spim_xfer()` from one place only, just like it was done before.

Reply:
Ok great catch again, I've addressed these items, tidied the PR and pushed again.  Hopefully getting close to complete now.  Please let me know any other issues you may see.